cromwell-backend/src/main/scala/cromwell/backend/model/ExecutionResult.scala

```diff
@@ -0,0 +1,22 @@
+package cromwell.backend.model
+
+import wdl4s.values.WdlValue
+
+/**
+  * Represents a result of task execution.
+  */
+trait ExecutionResult
+
+/**
+  * Succesful execution.
+  * @param outputs Outputs from task.
+  */
+case class SuccesfulResult(outputs: Map[String, WdlValue]) extends ExecutionResult
+
+/**
+  * Failure execution.
+  * @param exception Exception generated during task execution.
+  * @param rc Return code from command.
+  * @param stdErr Standard error data.
+  */
+case class FailureResult(exception: Throwable, rc: Option[Int], stdErr: String) extends ExecutionResult
```

Comment:
I'm pretty sure the Option comes from something which already exists in our codebase but to me when there's an `Option` in a product type like this it's a bit of a smell. What you really have are two separate forms of `FailureResult`, one with an `rc` and one without. You're already setting up `ExecutionResult` as a sum type, might as well split this into two product types.